lib/hawkular/operations/operations_api.rb

```diff
@@ -215,7 +215,10 @@ def check_pre_conditions(hash = {}, params = [], &callback)
       fail 'callback must have the perform method defined. include Hawkular::Operations' unless
           callback.nil? || callback.respond_to?('perform')
       params.each do |property|
-        fail "Hash property #{property} must be specified" if hash[property].nil?
+        if hash[property].nil?
+          error_message = "Hash property #{property} must be specified"
+          callback.nil? ? fail(error_message) : callback.perform(:failure, error_message)
```

Comment:
I guess that it would be better to raise an [ArgumentError](http://apidock.com/ruby/ArgumentError)